kubernetes/ExtServices.yml

```diff
@@ -0,0 +1,61 @@
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: fanout-ingress
+  annotations:
+    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
+    kubernetes.io/ingress.global-static-ip-name: external-ip-1
+  labels:
+    app: openwisp-nginx
+spec:
+  rules:
+  - host: dashboard.openwisp.org
+    http: &https_rules
+      paths:
+      - backend:
+          serviceName: openwisp-nginx
+          servicePort: 80
+        path: /*
+  - host: controller.openwisp.org
+    http: *https_rules
+  - host: radius.openwisp.org
+    http: *https_rules
+  - host: topology.openwisp.org
+    http: *https_rules
```

Comment:
Looks like this is in both ExtServices and Ingress.

Reply:
Yes, when I apply the "Ingress.yml" file on Google Cloud, it doesn't work and keeps complaining.
So, what I had to do is, create "ExtServices.yml" which doesn't work, it only creates an ingress and defines the path, then, I apply the "Certificates.yml" which gets me the letsencrypt certificates for me,
Finally, I apply the "Ingress.yml" which has the same block with
```
  tls:
  - hosts:
        ...
       (hosts)
        ...
    secretName: openwisp-2stacks-crt
``` 
So that I can finally use the site with https! :smile:

Reply:
I see what you mean with the circular references.  It seems like the step should be to;

1. Create the ingress without tls specification.
2. Create the certificate
3. Update the ingress with the tls specification.

That's pretty much what you're doing I guess but wonder if there's a way without duplicating so many lines of yaml.

Reply:
Indeed, that'll be ideal. Looking into the cert-manager docs for the same. :smile:

Reply:
Solved! :smile:

Reply:
Un-solve, the problem is that ingress-gce doesn't let ingress work if it has `tls` and no certificates.
So I had to a patch:
```
kubectl patch ingress/openwisp-http-ingress \
    --patch '{
        "spec": {
            "tls": [
                {
                "hosts": [
                    "dasboard.openwisp.org",
                    "controller.openwisp.org",
                    "radius.openwisp.org",
                    "topology.openwisp.org"
                ],
                "secretName": "openwisp-tls-secret"
                }
            ]
        }
    }'
```

There may be fix for this in future, so creating an issue